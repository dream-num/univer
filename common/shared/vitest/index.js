/**
 * Copyright 2023-present DreamNum Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const { defineConfig, mergeConfig } = require('vitest/config');

function createConfig(options) {
    return defineConfig(mergeConfig({
        test: {
            css: {
                modules: {
                    classNameStrategy: 'non-scoped',
                },
            },
            environment: 'happy-dom',
            coverage: {
                reporter: ['html', 'json'],
                provider: 'custom',
                customProviderModule: require.resolve('@vitest/coverage-istanbul'),
                exclude: [
                    'coverage/**',
                    'dist/**',
                    '**/[.]**',
                    'packages/*/test?(s)/**',
                    '**/*.d.ts',
                    '**/virtual:*',
                    '**/__x00__*',
                    '**/\x00*',
                    'cypress/**',
                    'test?(s)/**',
                    'test?(-*).?(c|m)[jt]s?(x)',
                    '**/*{.,-}{test,spec}?(-d).?(c|m)[jt]s?(x)',
                    '**/__test?(s)__/**',
                    '**/{karma,rollup,webpack,vite,vitest,jest,ava,babel,nyc,cypress,tsup,build}.config.*',
                    '**/vitest.{workspace,projects}.[jt]s?(on)',
                    '**/.{eslint,mocha,prettier}rc.{?(c|m)js,yml}',
                    'lib/**',
                    'src/locale/**',
                    '**/*.stories.tsx',
                    '**/__testing__/**',
                ],
            },
        },
    }, options));
}

module.exports = createConfig;
