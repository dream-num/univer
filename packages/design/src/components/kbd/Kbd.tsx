/**
 * Copyright 2023-present DreamNum Co., Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Fragment } from 'react/jsx-runtime';
import { borderClassName } from '../../helper/class-utilities';
import { clsx } from '../../helper/clsx';

export interface IKBDProps {
    keyboard: string;
    className?: string;
}

export function KBD(props: IKBDProps) {
    const { keyboard, className } = props;

    const keys = keyboard.split('+');

    return (
        <span
            className={clsx(`
              univer-inline-block univer-h-6 univer-select-none univer-rounded-md univer-bg-gray-50 univer-px-2
              univer-font-mono univer-text-xs/6 univer-font-medium univer-text-gray-700
              dark:univer-bg-gray-700 dark:univer-text-white
            `, borderClassName, className)}
        >
            {keys.map((text, index) => (
                <Fragment key={index}>
                    <kbd
                        className="univer-inline-block univer-h-full"
                    >
                        {text}
                    </kbd>
                    {index < keys.length - 1 && (
                        <span className="univer-inline-block univer-h-full univer-px-1">
                            +
                        </span>
                    )}
                </Fragment>
            ))}
        </span>
    );
}
