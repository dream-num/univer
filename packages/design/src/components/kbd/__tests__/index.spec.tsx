/**
 * Copyright 2023-present DreamNum Co., Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { cleanup, render } from '@testing-library/react';
import { afterEach, describe, expect, it } from 'vitest';
import { KBD } from '../Kbd';
import '@testing-library/jest-dom/vitest';

afterEach(cleanup);

describe('KBD', () => {
    it('should render with default props', () => {
        const { container } = render(<KBD keyboard="Ctrl" />);
        expect(container).toMatchSnapshot();
    });

    it('should render with custom className', () => {
        const { container } = render(<KBD className="custom-class" keyboard="Ctrl" />);
        expect(container.querySelector('.custom-class')).toBeTruthy();
    });

    it('should render with multiple keys', () => {
        const { container } = render(<KBD keyboard="Ctrl + Alt + Del" />);
        expect(container).toMatchSnapshot();
    });
});
