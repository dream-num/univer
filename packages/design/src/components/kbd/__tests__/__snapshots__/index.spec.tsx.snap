// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`KBD > should render with default props 1`] = `
<div>
  <span
    class="univer-inline-block univer-h-6 univer-select-none univer-rounded-md univer-bg-gray-50 univer-px-2 univer-font-mono univer-text-xs/6 univer-font-medium univer-text-gray-700 dark:!univer-bg-gray-700 dark:!univer-text-white univer-border-gray-200 dark:!univer-border-gray-600 univer-border-solid univer-border"
  >
    <kbd
      class="univer-inline-block univer-h-full"
    >
      Ctrl
    </kbd>
  </span>
</div>
`;

exports[`KBD > should render with multiple keys 1`] = `
<div>
  <span
    class="univer-inline-block univer-h-6 univer-select-none univer-rounded-md univer-bg-gray-50 univer-px-2 univer-font-mono univer-text-xs/6 univer-font-medium univer-text-gray-700 dark:!univer-bg-gray-700 dark:!univer-text-white univer-border-gray-200 dark:!univer-border-gray-600 univer-border-solid univer-border"
  >
    <kbd
      class="univer-inline-block univer-h-full"
    >
      Ctrl 
    </kbd>
    <span
      class="univer-inline-block univer-h-full univer-px-1"
    >
      +
    </span>
    <kbd
      class="univer-inline-block univer-h-full"
    >
       Alt 
    </kbd>
    <span
      class="univer-inline-block univer-h-full univer-px-1"
    >
      +
    </span>
    <kbd
      class="univer-inline-block univer-h-full"
    >
       Del
    </kbd>
  </span>
</div>
`;
